### DISCUSSION TITLE
The spatial resolution

### DISCUSSION THREAD

**Discussion by gsato-git** [FESOM/fesom2#631] [Q&A] [answered]
Hi. My name is Go Sato, and I am a PhD student. I would like to analyze the piControl and midHolocene SST data from the AWI-ESM model, but I cannot understand the coordinate system of this ocean component. Could you tell me how to specify the SST in latitude and longitude? For example, how can I plot the SST in Python in longitude and latitude at a specific moment?
Thank you!

**Comment by patrickscholz** [comment, marked as answer]
Hi Go Sato, the fesom2 repository has a folder named visualisation, in it are four suggestions what can be used to visualize fesom2 data. If you know python there is pyfesom2 (https://github.com/FESOM/pyfesom2, https://pyfesom2.readthedocs.io/en/stable/) and tripyview (https://github.com/FESOM/tripyview) that could help you to make your plots relatively easy. If you are next week at awi i could give you a brief introduction to tripyview

**Reply by gsato-git** [reply]
Thank you for answering my question. Could you give me a quick example of how to use the pfinterp function in pyfesom2? I read the documentation and tried this function many times while making some changes to the expression of my datapath and my meshpath, but somehow it doesn't work. Otherwise, do you have any recent updates on this function?
I currently work with AWI-ESM piControl output data from the ESGF website and mesh data from the GitHub FESOM2 repository.

**Reply by patrickscholz** [reply]
Oh I see  now, you are using AWI-ESM (v1) in that case the ocean model component is fesom1.4 and not fesom2. There are quite substantial differences between fesom1.4 and fesom2. To work with the original fesom1.4 data on a superficial level there is pyfesom (https://github.com/FESOM/pyfesom).  However if you use the CMIP6 data they might have been cmorized, so alternated from there original shape, where im not fully sure if pyfesom will work. Could you make an ...

 ncdump -h filename.nc

... and send me the result, to know with what format you deal. 

There should be also the possibilty to interpolate the fesom1.4 data using CDO, see here https://fesom.de/cmip6/work-with-awi-cm-unstructured-data/

**Reply by gsato-git** [reply]
I tried pyfesom, but because currently the swift.dkrz.de file, which has some basic files for interpolation, doesn't have proper mesh files for fesom, I can't process interpolation.

And below I attached my file format. This is the first file of the piControl simulation dataset.

netcdf tos_Omon_AWI-ESM-1-1-LR_piControl_r1i1p1f1_gn_185501-186012 {
dimensions:
        time = UNLIMITED ; // (72 currently)
        bnds = 2 ;
        ncells = 126859 ;
        vertices = 18 ;
variables:
        double time(time) ;
                time:standard_name = "time" ;
                time:long_name = "time" ;
                time:bounds = "time_bnds" ;
                time:units = "days since 1855-1-1 00:00:00" ;
                time:calendar = "standard" ;
                time:axis = "T" ;
        double time_bnds(time, bnds) ;
        float tos(time, ncells) ;
                tos:units = "degC" ;
                tos:CDI_grid_type = "unstructured" ;
                tos:_FillValue = 1.e+30f ;
                tos:missing_value = 1.e+30f ;
                tos:description = "Temperature of upper boundary of the liquid ocean, including temperatures below sea-ice and floating ice shelves." ;
                tos:coordinates = "lat lon" ;
                tos:standard_name = "sea_surface_temperature" ;
                tos:cell_methods = "area: mean where sea time: mean" ;
                tos:cell_measures = "area: areacello" ;
        double lat(ncells) ;
                lat:units = "degrees_north" ;
                lat:standard_name = "latitude" ;
                lat:bounds = "lat_bnds" ;
        double lat_bnds(ncells, vertices) ;
        double lon(ncells) ;
                lon:units = "degrees_east" ;
                lon:standard_name = "longitude" ;
                lon:bounds = "lon_bnds" ;
        double lon_bnds(ncells, vertices) ;
// global attributes:
                :frequency = "mon" ;
                :activity_id = "CMIP" ;
                :Conventions = "CF-1.7 CMIP-6.2" ;
                :creation_date = "2020-02-12T12:00:00Z" ;
                :data_specs_version = "01.00.30" ;
                :experiment = "piControl" ;
                :experiment_id = "piControl" ;
                :forcing_index = 1 ;
                :further_info_url = "https://furtherinfo.es-doc.org/CMIP6.AWI.AWI-ESM-1-1-LR.piControl.none.r1i1p1f1" ;
                :grid = "FESOM 1.4 (unstructured grid in the horizontal with 126859 wet nodes; 46 levels; top grid cell 0-5 m)" ;
                :grid_label = "gn" ;
                :initialization_index = 1 ;
                :institution = "Alfred Wegener Institute, Helmholtz Centre for Polar and Marine Research, Am Handelshafen 12, 27570 Bremerhaven, Germany" ;
                :institution_id = "AWI" ;
                :license = "CMIP6 model data produced by Alfred Wegener Institute, Helmholtz Centre for Polar and Marine Research, Am Handelshafen 12, 27570 Bremerhav
en, Germany is licensed under a Creative Commons Attribution-ShareAlike 4.0 International License (https://creativecommons.org/licenses/). Consult https://pcmdi.llnl.
gov/CMIP6/TermsOfUse for terms of use governing CMIP6 output, including citation requirements and proper acknowledgment. Further information about this data, includin
g some limitations, can be found via the further_info_url (recorded as a global attribute in this file). The data producers and data providers make no warranty, eithe
r express or implied, including, but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of th
e information (including any liability arising in negligence) are excluded to the fullest extent permitted by law." ;
                :mip_era = "CMIP6" ;
                :nominal_resolution = "50 km" ;
                :physics_index = 1 ;
                :product = "model-output" ;
                :realization_index = 1 ;
                :realm = "ocean" ;
                :source = "AWI-ESM-1-1-LR" ;
                :source_id = "AWI-ESM-1-1-LR" ;
                :source_type = "AOGCM" ;
                :sub_experiment = "none" ;
                :sub_experiment_id = "none" ;
                :table_id = "Omon" ;
                :tracking_id = "hdl:21.14100/e294ad0f-6569-454e-919f-380a8b230a3b" ;
                :variable_id = "tos" ;
                :variant_label = "r1i1p1f1" ;
                :branch_method = "standard" ;
                :branch_time_in_child = 0. ;
                :branch_time_in_parent = 617262. ;
                :parent_activity_id = "CMIP" ;
                :parent_experiment_id = "piControl-spinup" ;
                :parent_mip_era = "CMIP6" ;
                :parent_source_id = "AWI-ESM-1-1-LR" ;
                :parent_time_units = "days since 1-1-1" ;
                :parent_variant_label = "r1i1p1f1" ;
}

**Comment by patrickscholz** [comment]
So indeed i think you deal with cmorized cmip fesom14 data! The corresponding mesh files that belong to that data you could be found in /pool/data/AWICM/FESOM1/MESHES . The mesh you used here should be "core". 
So in theory you could use tripyview in that case, since the data structure is relatively close to fesom2. In https://github.com/FESOM/tripyview/blob/main/tools/template_cmip6_transp_zmoc.ipynb i wrote a fast script for somebody else to compute the AMOC based on CMIP6 fesom1.4 data using the tripyview fesom2 infrastructure. I could do something similar for you, at least for a limited number of analysis templates (see tripyview/template_notebooks).

**Comment by patrickscholz** [comment]
In the levante folder /pool/data/AWICM/FESOM1/MESHES/core you also find the griddes.nc file that is needed to do the interpolation to regular grid using CDO

Here is also detailed description how to do the interpolation to regular grid with AWI-ESM for CMIP6 data https://fesom.de/cmip6/work-with-awi-cm-unstructured-data/

**Reply by gsato-git** [reply]
I could do the interpolation to regular grid using cdo command! 

Could you share the folder /pool/data/AWICM/FESOM1/MESHES? I can't find it...

**Reply by patrickscholz** [reply]
You do have access to DKRZ levante (levante.dkrz.de), right? 

There its the folder

/pool/data/AWICM/FESOM1/MESHES/core/

**Comment by gsato-git** [comment]
I only have swift.dkrz.de which I got here; https://github.com/FESOM/pyfesom/blob/master/notebooks/get_data.ipynb.

**Comment by patrickscholz** [comment]
Hi Go Sato

I just checked a bit, you dont need the extra fesom1.4 mesh information to be able to interpolate the fesom14CMIP6 data to a regular grid. These datas already contain lon/lat information that is necessary for the interpolation. All you have to do is using CDO (Climate Data Operator)  ...

`cdo remapycon,global_1 tos_Omon_AWI-ESM-1-1-LR_piControl_r1i1p1f1_gn_185501-186012.nc 1x1deg.nc`

... this will conservatively interpolate your data to a 1x1deg regular grid (see here: https://fesom.de/cmip6/work-with-awi-cm-unstructured-data/). You can also store the interpolation weights extra and reuse them for interpolation with ...

`cdo genycon,global_1 tos_Omon_AWI-ESM-1-1-LR_piControl_r1i1p1f1_gn_185501-186012.nc weights_1x1`
`cdo remap,global_1,weights_1x1 tos_Omon_AWI-ESM-1-1-LR_piControl_r1i1p1f1_gn_185501-186012.nc 1x1deg.nc`

With cdo you can also do all kind of averages, regional selctions etc. ...

As long as you stick to scalar quantities interpolation is fine, but if you also want to analyze horizontal transports and fluxes than you should do as much as you can on the native grid. Because interpolation might alternate the outcome to a certain degree.

However if you really want to work with the native fesom14 unstructured grid and you want to plot the unstructured grid, than you need the original grid files. But these grid files you can also find on that dkrz swift browser: https://swiftbrowser.dkrz.de/public/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom-data/COREII/. Than you could use tripyview  e.g. https://github.com/FESOM/tripyview/blob/main/tools/fesom14cmip6/template_cmip6_hslice.ipynb to plot your data as shown in the example.

**Reply by gsato-git** [reply]
I could interpolate the data to regular grid using the cdo command! Now I'm focusing on only SST, but I will try to work with original fesom1.4 unstructured grid to investigate fluxes. Once again, thank you!
